examples/hal_lib/makefile-common.mk

```diff
@@ -0,0 +1,132 @@
+SDCARD_MOUNT_PATH ?= /Volumes/BAREAPP
+
+LINKSCR ?= linkscript.ld
+EXTLIBDIR ?= ../../third-party
+UBOOTDIR ?= $(EXTLIBDIR)/u-boot/build
+BUILDDIR ?= build
+BINARYNAME ?= main
+UIMAGENAME ?= $(BUILDDIR)/a7-main.uimg
+
+OBJDIR = $(BUILDDIR)/obj
+LOADADDR 	= 0xC2000040
+ENTRYPOINT 	= 0xC2000040
+
+OBJECTS   = $(addprefix $(OBJDIR)/, $(addsuffix .o, $(basename $(SOURCES))))
+DEPS   	  = $(addprefix $(OBJDIR)/, $(addsuffix .d, $(basename $(SOURCES))))
+
+MCU ?=  -mcpu=cortex-a7 -march=armv7ve -mfpu=neon-vfpv4 -mlittle-endian -mfloat-abi=hard
+
+ARCH_CFLAGS ?= -DUSE_FULL_LL_DRIVER \
+			  -DSTM32MP157Cxx \
+			  -DSTM32MP1 \
+			  -DCORE_CA7 \
+
+OPTFLAG ?= -O0
+
+AFLAGS = $(MCU)
+
+CFLAGS = -g2 \
+		 -fno-common \
+		 $(ARCH_CFLAGS) \
+		 $(MCU) \
+		 $(INCLUDES) \
+		 -fdata-sections -ffunction-sections \
+		 -nostartfiles \
+		 -ffreestanding \
+		 $(EXTRACFLAGS)\
+
+CXXFLAGS = $(CFLAGS) \
+		-std=c++2a \
+		-fno-rtti \
+		-fno-exceptions \
+		-fno-unwind-tables \
+		-ffreestanding \
+		-fno-threadsafe-statics \
+		-Werror=return-type \
+		-Wdouble-promotion \
+		-Wno-register \
+		 $(EXTRACXXFLAGS) \
+
+LFLAGS = -Wl,--gc-sections \
+	-Wl,-Map,$(BUILDDIR)/$(BINARYNAME).map,--cref \
+	$(MCU)  \
+	-T $(LINKSCR) \
+	-nostartfiles \
+	-ffreestanding \
+	$(EXTRALDFLAGS) \
+
+DEPFLAGS = -MMD -MP -MF $(OBJDIR)/$(basename $<).d
+
+ARCH 	= arm-none-eabi
+CC 		= $(ARCH)-gcc
+CXX 	= $(ARCH)-g++
+LD 		= $(ARCH)-g++
+AS 		= $(ARCH)-as
+OBJCPY 	= $(ARCH)-objcopy
+OBJDMP 	= $(ARCH)-objdump
+GDB 	= $(ARCH)-gdb
+SZ 		= $(ARCH)-size
+
+SZOPTS 	= -d
+
+ELF 	= $(BUILDDIR)/$(BINARYNAME).elf
+HEX 	= $(BUILDDIR)/$(BINARYNAME).hex
+BIN 	= $(BUILDDIR)/$(BINARYNAME).bin
+
+all: Makefile $(ELF) $(UIMAGENAME)
+
+install:
+	cp $(UIMAGENAME) $(SDCARD_MOUNT_PATH)
+	diskutil unmount $(SDCARD_MOUNT_PATH)
+
+$(OBJDIR)/%.o: %.s
+	@mkdir -p $(dir $@)
+	$(info Building $< at $(OPTFLAG))
+	@$(AS) $(AFLAGS) $< -o $@ 
+
+$(OBJDIR)/%.o: %.c $(OBJDIR)/%.d
+	@mkdir -p $(dir $@)
+	$(info Building $< at $(OPTFLAG))
+	@$(CC) -c $(DEPFLAGS) $(OPTFLAG) $(CFLAGS) $< -o $@
+
+$(OBJDIR)/%.o: %.cc $(OBJDIR)/%.d
+	@mkdir -p $(dir $@)
+	$(info Building $< at $(OPTFLAG))
+	@$(CXX) -c $(DEPFLAGS) $(OPTFLAG) $(CXXFLAGS) $< -o $@
+
+$(OBJDIR)/%.o: %.cpp $(OBJDIR)/%.d
```

Comment:
Did my Makefile regex attempt not work for you? I had this originally:
```
$(OBJDIR)/%.o: %.c[cp]* $(OBJDIR)/%.d
```
I'm not totally happy with what I had, since it would match files other than .cc and .cpp (e.g. .ccp or .cpcp). Did you get an error? I'm OK with switching back to having two separate rules for .cc and .cpp if it makes it more compatible with various versions of make